src/server/pps/server/api_server.go

```diff
@@ -54,6 +58,37 @@ type apiServer struct {
 
 func (a *apiServer) CreateJob(ctx context.Context, request *ppsclient.CreateJobRequest) (response *ppsclient.Job, retErr error) {
 	defer func(start time.Time) { a.Log(request, response, retErr, time.Since(start)) }(time.Now())
+	persistClient, err := a.getPersistClient()
+	if err != nil {
+		return nil, err
+	}
+
+	if request.Pipeline != nil {
+		var inputCommits []*pfsclient.Commit
+		for _, input := range request.Inputs {
+			inputCommits = append(inputCommits, input.Commit)
+		}
+
+		jobInfos, err := persistClient.ListJobInfos(ctx, &ppsclient.ListJobRequest{
+			Pipeline:    request.Pipeline,
+			InputCommit: inputCommits,
+		})
+
+		if err != nil {
+			return nil, err
+		}
+
+		if len(jobInfos.JobInfo) > 1 {
+			return nil, fmt.Errorf("there are %d jobs that match the job info (%v); this is likely a bug", len(jobInfos.JobInfo), request)
+		}
+
+		if len(jobInfos.JobInfo) != 0 {
+			return &ppsclient.Job{
+				ID: jobInfos.JobInfo[0].JobID,
+			}, nil
+		}
+	}
```

Comment:
Hmm, so I just realized it could still be possible for 2 identical jobs to get created here if both calls check for the other's job at exactly the same time.

Maybe we should make it so the id for a job id is a hash of the pipeline and inputs, then we can just call insert and it will error if you try to insert 2 identical jobs.